kafka-connect/kafka-connect/src/main/java/org/apache/iceberg/connect/data/IcebergWriterFactory.java

```diff
@@ -112,4 +117,18 @@ Table autoCreateTable(String tableName, SinkRecord sample) {
             });
     return result.get();
   }
+
+  @VisibleForTesting
+  static void createNamespaceIfNotExist(Catalog catalog, Namespace identifierNamespace) {
+    String[] levels = identifierNamespace.levels();
+    for (int index = 0; index < levels.length; index++) {
+      Namespace namespace = Namespace.of(Arrays.copyOfRange(levels, 0, index + 1));
+      try {
+        ((SupportsNamespaces) catalog).createNamespace(namespace);
+      } catch (AlreadyExistsException ex) {
```

Comment:
We went the other way with Flink: https://github.com/apache/iceberg/pull/7795

Can we also swallow the exception where the user doesn't have permission to create a namespace.

Reply:
I assumed if the auto create table is enabled, means user has permission to create tables. So, They can create Namespaces also. But that assumptions can be wrong for some catalogs. Let me catch `ForbiddenException` also for ignoring it.